app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer.go

```diff
@@ -59,13 +60,24 @@ var _ ForwardingRuleSyncerInterface = &ForwardingRuleSyncer{}
 // Does nothing if it exists already, else creates a new one.
 // Stores the given list of clusters in the description field of forwarding rule to use it to generate status later.
 func (s *ForwardingRuleSyncer) EnsureHttpForwardingRule(lbName, ipAddress, targetProxyLink string, clusters []string, forceUpdate bool) error {
-	fmt.Println("Ensuring http forwarding rule")
-	desiredFR, err := s.desiredForwardingRule(lbName, ipAddress, targetProxyLink, clusters)
+	s.namer.HttpForwardingRuleName()
+	return s.ensureForwardingRule(lbName, ipAddress, targetProxyLink, httpDefaultPortRange, "http", s.namer.HttpForwardingRuleName(), clusters, forceUpdate)
+}
+
+// EnsureHttpsForwardingRule ensures that the required https forwarding rule exists.
+// Does nothing if it exists already, else creates a new one.
+// Stores the given list of clusters in the description field of forwarding rule to use it to generate status later.
+func (s *ForwardingRuleSyncer) EnsureHttpsForwardingRule(lbName, ipAddress, targetProxyLink string, clusters []string, forceUpdate bool) error {
+	return s.ensureForwardingRule(lbName, ipAddress, targetProxyLink, httpsDefaultPortRange, "https", s.namer.HttpsForwardingRuleName(), clusters, forceUpdate)
+}
+
+func (s *ForwardingRuleSyncer) ensureForwardingRule(lbName, ipAddress, targetProxyLink, portRange, httpProtocol, name string, clusters []string, forceUpdate bool) error {
```

Comment:
should this have a function comment?

Reply:
As a rule, private methods are not required to.
Have added a comment here since it has significant code.